diff --git a/PKGBUILD b/PKGBUILD
index 69f3249..760b820 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,7 +27,7 @@ prepare() {
 	dub add-local $srcdir/i2d-imgui/ "0.8.0"
 
 	# Unofficial builds required to change bug report URL
-	sed -i "s,https://github.com/Inochi2D/inochi-creator/issues/new?assignees=&labels=bug&template=bug-report.yml&title=%5BBUG%5D,https://aur.archlinux.org/packages/inochi-creator,g" source/creator/config.d
+	sed -i "s,https://github.com/Inochi2D/inochi-creator/issues/new?assignees=&labels=bug&template=bug-report.yml&title=%5BBUG%5D,https://github.com/akku1139/auto-aur/issues/new?labels=bug&title=[inochi-creator]%20,g" source/creator/config.d
 }
 
 build() {
